src/session.rs

```diff
@@ -1002,6 +1027,13 @@ impl Session {
     fn inner(&self) -> MutexGuard<SessionInner> {
         self.inner.lock()
     }
+
+    /// Sets the trace level for the session.
+    ///
+    pub fn trace(&self, bitmask: TraceFlags) {
+        let inner = self.inner();
+        unsafe { let _ = raw::libssh2_trace(inner.raw, bitmask.bits() as c_int); }
```

Comment:
Are you sure this have to be unsafe? I think it should not but maybe I am missing something here.

Reply:
The `unsafe` here is required because it's an `extern "C"` ffi function and Rust automatically marks those as unsafe.

Reply:
Yeap, for some reason I though this was a wrapper like when we open a session and not a direct call to the underlying C companion.